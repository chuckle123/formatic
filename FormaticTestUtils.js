import React from 'react';
import { mount } from 'enzyme';

import Formatic from './lib/formatic';

export const renderFieldsToHtml = fields => {
  const config = Formatic.createConfig();
  const value = config.createRootValue({
    fields,
  });

  const component = mount(
    <Formatic
      config={config}
      value={value}
      fields={fields}
      onChange={() => {}}
    />
  );
  const html = component.html();
  component.unmount();
  return html;
};
