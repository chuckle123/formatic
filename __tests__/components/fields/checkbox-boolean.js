/*global describe, it, expect*/

import { renderFieldsToHtml } from '../../../FormaticTestUtils';
import checkboxBooleanExample from '../../../demo/examples/checkbox-boolean';

describe('checkbox-boolean field', () => {
  it('should render correctly', () => {
    expect(renderFieldsToHtml(checkboxBooleanExample.fields)).toMatchSnapshot();
  });
});
