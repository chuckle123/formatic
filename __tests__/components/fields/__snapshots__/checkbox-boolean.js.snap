// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox-boolean field should render correctly 1`] = `

<div class="formatic">
  <div class="optional">
    <div class>
      <span class="required-label not-required-text">
      </span>
    </div>
    <span>
    </span>
    <fieldset class>
      <span style="white-space: nowrap;">
        <input type="checkbox"
               class
               disabled
        >
        <span>
        </span>
        <span>
          Enable Read-only
        </span>
      </span>
      <span style="white-space: nowrap;">
        <input type="checkbox"
               class
        >
        <span>
        </span>
        <span>
          Is this checked?
        </span>
      </span>
      <span style="white-space: nowrap;">
        <input type="checkbox"
               class
        >
        <span>
        </span>
        <span>
          Is this a checkbox?
        </span>
      </span>
    </fieldset>
  </div>
</div>

`;
